### PR TITLE
Add daily papers endpoint

### PR DESCRIPTION
**Pull request by BastienGimbert**
# Problem :
 There was no simple function to get daily papers by date from the API.

# Summary :
Add HfApi.list_daily_papers(date) to call /api/daily_papers.
Add a small test to check the endpoint returns objects and ISO date

# Testing
pytest -q tests/test_hf_api.py::PaperApiTest

### CODE REVIEW

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -9702,6 +9702,41 @@ def paper_info(self, id: str) -> PaperInfo:
         hf_raise_for_status(r)
         return PaperInfo(**r.json())
 
+    def list_daily_papers(
+        self,
+        *,
+        date: str,
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py]:
```suggestion
        date: Optional[str] = None,
        *,
```

Date should be made optional, in which case it returns recent ones https://huggingface.co/api/daily_papers

(also it's fine to have `date` as positional arg)

File: src/huggingface_hub/hf_api.py [hunk 2]
```diff
@@ -9702,6 +9702,41 @@ def paper_info(self, id: str) -> PaperInfo:
...
+        token: Union[bool, str, None] = None,
+    ) -> Iterable[PaperInfo]:
+        """
+        List the daily papers published on a given date on the Hugging Face Hub.
+
+        Args:
+            date (`str`):
+                Date in ISO format (YYYY-MM-DD) for which to fetch daily papers.
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py:9716]:
```suggestion
                Date in ISO format (YYYY-MM-DD) for which to fetch daily papers.
                Defaults to most recent ones.
```

File: src/huggingface_hub/hf_api.py [hunk 3]
```diff
@@ -9702,6 +9702,41 @@ def paper_info(self, id: str) -> PaperInfo:
...
+
+        ```python
+        >>> from huggingface_hub import HfApi
+
+        >>> api = HfApi()
+        >>> list(api.list_daily_papers(date="2025-10-29"))
+        ```
+        """
+        path = f"{self.endpoint}/api/daily_papers"
+        params = {"date": date}
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py]:
```suggestion
        params = {"date": date} if date is not None else {}
```

File: src/huggingface_hub/hf_api.py [hunk 4]
```diff
@@ -10723,6 +10758,7 @@ def _parse_revision_from_pr_url(pr_url: str) -> str:
 
 list_papers = api.list_papers
 paper_info = api.paper_info
+list_daily_papers = api.list_daily_papers
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py:10762]: must be added to the main `__init__.py` file as well

File: tests/test_hf_api.py [hunk 5]
```diff
@@ -4228,6 +4229,16 @@ def test_get_paper_by_id_not_found(self) -> None:
...
         assert context.exception.response.status_code == 404
 
+    def test_list_daily_papers_by_date(self) -> None:
+        papers = list(self.api.list_daily_papers(date="2025-10-29"))
+        assert len(papers) > 0
+        assert hasattr(papers[0], "id")
+        assert hasattr(papers[0], "title")
+
+    def test_list_daily_papers_by_date_invalid_date(self) -> None:
+        with self.assertRaises(ValueError):
```
Comment (Wauplin) [inline tests/test_hf_api.py]:
```suggestion
        with self.assertRaises(BadRequestError):
```

more specific to raise BadRequestError (HTTP 400)

File: tests/test_hf_api.py [hunk 6]
```diff
@@ -4228,6 +4229,16 @@ def test_get_paper_by_id_not_found(self) -> None:
             self.api.paper_info("1234.56789")
         assert context.exception.response.status_code == 404
 
+    def test_list_daily_papers_by_date(self) -> None:
```
Comment (Wauplin) [inline tests/test_hf_api.py:4231]: let's add another test for when date is None

**Comment by BastienGimbert** [comment]
Hello @Wauplin,
I made all the changes you asked me to and added the new test.
6/6 tests passed
And yes, I forgot to declare it in ‘__init__.py’. my bad ;')

Have a nice day !

File: tests/test_hf_api.py [hunk 7]
```diff
@@ -64,6 +64,7 @@
     SpaceInfo,
     SpaceRuntime,
     User,
+    PaperInfo,
```
Comment (Wauplin) [inline tests/test_hf_api.py]:
```suggestion
```

File: src/huggingface_hub/hf_api.py [hunk 8]
```diff
@@ -9702,6 +9702,42 @@ def paper_info(self, id: str) -> PaperInfo:
...
+        self,
+        *,
+        date: Optional[str] = None,
+        token: Union[bool, str, None] = None,
+    ) -> Iterable[PaperInfo]:
+        """
+        List the daily papers published on a given date on the Hugging Face Hub.
+
+        Args:
+            date (`str`):
```
Comment (Samoed) [inline src/huggingface_hub/hf_api.py]:
```suggestion
            date (`str`, *optional*):
```

**Comment by Wauplin** [comment]
@bot /style

**Comment by Wauplin** [comment]
> Seems from https://huggingface-openapi.hf.space/#tag/papers/get/apipaperssearch daily papers also support p, limit, week, month, sort fields

@Samoed  Would you like to open a follow-up PR once this one is merged to add support for these parameters? :)

**Comment by Samoed** [comment]
Yes, of course
